csharp/src/Apache.Arrow/ArrowBuffer.BitmapBuilder.cs

```diff
@@ -89,6 +89,37 @@ public BitmapBuilder Append(bool value)
                 return this;
             }
 
+            /// <summary>
+            /// Append a span of bits.
+            /// </summary>
+            /// <param name="source">Source of bit span.</param>
+            /// <param name="validBits">Number of valid bits in the bit span</param>
+            /// <returns>Returns the builder (for fluent-style composition).</returns>
+            public BitmapBuilder Append(ReadOnlySpan<byte> source, int validBits)
+            {
+                var length = source.IsEmpty ? validBits : Math.Min(source.Length * 8, validBits);
```

Comment:
Since this is a public API, I think we should we do parameter validation that ensures `validBits` <= `source.Length * 8`. Or we make it internal.

Reply:
Fixed